test/jdk/tools/jpackage/helpers/jdk/jpackage/test/LauncherAsServiceVerifier.java

```diff
@@ -242,22 +253,40 @@ private void applyToMainLauncher(PackageTest pkg) {
     }
 
     private void applyToAdditionalLauncher(PackageTest pkg) {
-        new AdditionalLauncher(launcherName) {
+        AdditionalLauncher al = new AdditionalLauncher(launcherName) {
             @Override
             protected void verify(JPackageCommand cmd) throws IOException {
                 if (canVerifyInstall(cmd)) {
                     delayInstallVerify();
                     super.verify(cmd);
-                    TKit.deleteIfExists(appOutputFilePathVerify(cmd));
+                    deleteOutputFile(appOutputFilePathVerify(cmd));
                 }
                 LauncherAsServiceVerifier.verify(cmd, launcherName);
             }
         }.setLauncherAsService()
                 .addJavaOptions("-Djpackage.test.appOutput="
                         + appOutputFilePathInitialize().toString())
                 .addJavaOptions("-Djpackage.test.noexit=true")
-                .addDefaultArguments(expectedValue)
-                .applyTo(pkg);
+                .addDefaultArguments(expectedValue);
+
+        Optional.ofNullable(additionalLauncherCallback).ifPresent(v -> v.accept(al));
+
+        al.applyTo(pkg);
+    }
+
+    private static void deleteOutputFile(Path file) throws IOException {
+        try {
+            TKit.deleteIfExists(file);
+        } catch (FileSystemException ex) {
+            if (TKit.isLinux() || TKit.isOSX()) {
+                // Probably "Operation no permitted" error. Try with "sudo" as the
+                // file is created by a launcher started under root account.
+                Executor.of("sudo", "rm", "-f").addArgument(file.toString()).
+                        execute();
+            } else {
+                throw ex;
+            }
+        }
```

Comment:
Fixed an issue that made it impossible to run tests for the "launcher as service" feature without starting jtreg under `sudo`.